Update release guide (#8932)
Scille/parsec-cloud: docs/development/guide-release-parsec.md

With the release published, this will trigger the workflow `publish` that will upload the generated python wheels to `pypi`.

### Continue stabilizing the release

When we prepare a release we first mark-it as a pre-release denoted by the release candidate part of the version (the `-rc.N` thingy).

> Once a pre-release is publish, you perform some tests on it.

Reviewer: ```suggestion
Once a pre-release is published, it goes through a test campaign to assess its stability.
```

If the tests suite reveal that the current release candidate is stable, you can go to next step [Once the release is stable](#once-the-release-is-stable)

Else, you will need to add fixes to the release branch (cf [Preparing the patched version](#preparing-the-patched-version)) and create the next release candidate.

To create the next RC once you've added the fixes you wanted, you execute the following command:

```shell
python misc/releaser.py build --current
```

> [!NOTE]
> We don't specify the version here but use the `--current` flag as the current version of the release branch at this step is `v3.2.0-rc.1+dev`

Like in [create the release](#create-the-release), follow the steps in [after the release build script](#after-the-release-build-script)

### Once the release is stable

Once the release candidate is considered stable and ready for a stable (that the public can use) release

```shell
python misc/releaser.py build --version 3.2.0
```

> Then follow the remaining step described in [after the release build script](#after-the-release-build-script)

Reviewer: ```suggestion
Then follow the remaining steps described in [after the release build script](#after-the-release-build-script)
```

## Release patch version

In the following we will consider we want to release version ``v3.2.1`` from `v3.2.1-a.0+dev`.

### (Re)create the version branch

If the release branch ``releases/3.2`` used for the ``3.2.0`` release has been
removed, it needs to be recreated.

> [!IMPORTANT]
> The release branch should live on their own and should not be deleted

In that case, use the git tag `v3.2.0` and push it to the branch `releases/3.2`

Of course the version branch should be reused if a previous patch release has
already been done (e.g. you're planning to release ``v3.2.2``).

### Preparing the patched version

> To include some fixes for the patch version, you can do that in 2 different way:

Reviewer: ```suggestion
To include some fixes for the patch version, you can do that in 2 different ways:
```
